File: src/test/java/ValidPalindromeTest.java
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class ValidPalindromeTest {
    @ParameterizedTest
    @MethodSource("testCases")
    public void testValidPalindrome(String s, boolean expected){

        boolean result = ValidPalindrome.isPalindrome(s);

        assertEquals(expected, result);
    }

    static Stream<Arguments> testCases(){
        return Stream.of(
                Arguments.of(" ", true),
                Arguments.of("race a car", false),
                Arguments.of("A man, a plan, a canal: Panama", true)
        );
    }
}

File: leetCode/test/ValidPalindromeTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using leetCode;

namespace test
{
    [TestClass]
    public class ValidPalindromeTest
    {
        [DataTestMethod]
        [DataRow("A man, a plan, a canal: Panama", true)]
        [DataRow("race a car", false)]
        [DataRow(" ", true)]
        public void IsPalindromeTest(string s, bool expected) 
        {
            bool res = ValidPalindrome.IsPalindrome(s);

            Assert.AreEqual(res, expected); 
        }
    }
}
